free5gc-operator/controllers/smfdeployment_controller.go

```diff
@@ -0,0 +1,648 @@
+		            sd: 112234
+		          dnnInfos:
+		            - dnn: internet
+		              dns:
+		                ipv4: 8.8.8.8
+		      plmnList: # the list of PLMN IDs that this SMF belongs to (optional, remove this key when unnecessary)
+		        - mcc: "208" # Mobile Country Code (3 digits string, digit: 0~9)
+		          mnc: "93" # Mobile Network Code (2 or 3 digits string, digit: 0~9)
+		      userplaneInformation: # list of userplane information
+		        upNodes: # information of userplane node (AN or UPF)
+		          gNB1: # the name of the node
+		            type: AN # the type of the node (AN or UPF)
+		          UPF:  # the name of the node
+		            type: UPF # the type of the node (AN or UPF)
+		            nodeID: 10.100.50.241 # the IP/FQDN of N4 interface on this UPF (PFCP)
+		            sNssaiUpfInfos: # S-NSSAI information list for this UPF
+		                  - sNssai: # S-NSSAI (Single Network Slice Selection Assistance Information)
+		                      sst: 1 # Slice/Service Type (uinteger, range: 0~255)
+		                      sd: 010203 # Slice Differentiator (3 bytes hex string, range: 000000~FFFFFF)
+		                    dnnUpfInfoList: # DNN information list for this S-NSSAI
+		                      - dnn: internet
+		                        pools:
+		                          - cidr: 10.1.0.0/17
+		                  - sNssai: # S-NSSAI (Single Network Slice Selection Assistance Information)
+		                      sst: 1 # Slice/Service Type (uinteger, range: 0~255)
+		                      sd: 112233 # Slice Differentiator (3 bytes hex string, range: 000000~FFFFFF)
+		                    dnnUpfInfoList: # DNN information list for this S-NSSAI
+		                      - dnn: internet
+		                        pools:
+		                          - cidr: 10.1.128.0/17
+		            interfaces: # Interface list for this UPF
+		                  - interfaceType: N3 # the type of the interface (N3 or N9)
+		                    endpoints: # the IP address of this N3/N9 interface on this UPF
+		                      - 10.100.50.233
+		                    networkInstance: internet # Data Network Name (DNN)
+		        links: # the topology graph of userplane, A and B represent the two nodes of each link
+		          - A: gNB1
+		            B: UPF
+		      locality: area1 # Name of the location where a set of AMF, SMF and UPFs are located
+
+		    logger:
+		      Aper:
+		        ReportCaller: false
+		        debugLevel: info
+		      NAS:
+		        ReportCaller: false
+		        debugLevel: info
+		      NGAP:
+		        ReportCaller: false
+		        debugLevel: info
+		      PFCP:
+		        ReportCaller: false
+		        debugLevel: info
+		      SMF:
+		        ReportCaller: false
+		        debugLevel: info
+	*/
+	smfcfgTemplate := template.New("SMFCfg")
+	smfcfgTemplate, err = smfcfgTemplate.Parse(SMFCfgTemplate)
+	if err != nil {
+		log.Log.Info("Could not parse SMFCfgTemplate template.")
```

Comment:
for all these don't log, just wrap the error if you need to add the context of which template you failed to parse. See https://go.dev/blog/go1.13-errors for a discussion, in particular the "Wrapping errors with %w" section.

TL;DR, instead of log and return, just do:
```go
return nil, fmt.Errorf("SMFCfgTemplate parse error: %w", err)
```